src/order/interval.lean

```diff
@@ -0,0 +1,125 @@
+/-
+Copyright (c) 2022 Yaël Dillies. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies
+-/
+import data.set.intervals.basic
+
+/-!
+# Order intervals
+
+This file defines intervals in an order as pairs where the second element is greater than the first.
+This is a prototype for interval arithmetic.
+-/
+
+open function order_dual set
+
+/-- The intervals in an order. -/
+@[ext] structure interval (α : Type*) [has_le α] extends α × α :=
+(fst_le_snd : fst ≤ snd)
+
+namespace interval
+variables {α β γ : Type*}
+
+section has_le
+variables [has_le α] {a b : interval α}
+
+/-- The injection from which we get the order. -/
+def to_dual_prod : interval α → αᵒᵈ × α := interval.to_prod
```

Comment:
This abuses defeq of order_dual, right? Do you want to define it using `to_dual` somewhere, or have a lemma `to_dual_prod_apply` whose right hand side uses `to_dual`?